sympy/solvers/bivariate.py

```diff
@@ -151,7 +154,13 @@ def _lambert(eq, x):
 
     # invert the generator X1 so we have x(u)
     u = Dummy('rhs')
-    xusolns = solve(X1 - u, x)
+    if domain is None:
+        # the flow from solve will proceed from here
+        domain_real = None
+        xusolns = solve(X1 - u, x)
+    else:
+        domain_real = domain.is_subset(S.Reals)
+        xusolns = solvify(X1 - u, x, domain)
```

Comment:
>  I still find looking at inequalities.py that I don't understand why so many things have changed when this PR shouldn't have anything to do with inequalities. 

Please try to understand that `solvify` is being used and is common for both and the `solvify` functions is also modified in the way so that it should give the solutions to `intersection` set also....